src/textual/widget.py

```diff
@@ -803,16 +803,16 @@ def watch_hover_style(
             self.highlight_link_id = hover_style.link_id
 
     def watch_scroll_x(self, old_value: float, new_value: float) -> None:
-        if self.show_horizontal_scrollbar:
+        if self.allow_horizontal_scroll:
             self.horizontal_scrollbar.position = round(new_value)
```

Comment:
I was reading your comment in the original issue and I'm not convinced `allow_horizontal_scroll` should be used here. Note that `scroll_x` was changed, which is why we are in `watch_scroll_x`, so now that `scroll_x` was changed, we need to update the horizontal scrollbar if it is visible.

To me, scrollbar being visible != scrolling enabled, so we really want to check if we are _showing_ the scrollbar.

Reply:
Hasn't the logic already been done in the various scroll methods, with regards to allowing scrolling?

By the time we get here, I don't think we need to apply the logic again.

This does mean that you can circumvent scroll logic by setting the `scroll_x`  attribute, but I think I'm fine with that.

Reply:
Makes sense to me, I only made that change as it was something you explicitly asked for. I'll nuke that then.

Reply:
Done.